src/wallet_client.ts

```diff
@@ -208,16 +212,32 @@ export class WalletClient {
     return await this.provider.requestSuiFromFaucet(address);
   }
 
-  async getCoinsWithRequiredBalance(address: string, amount: number) {
+  async getCoinsWithRequiredBalance(address: string, amount: number, typeArg: string = SUI_TYPE_ARG) {
     const coinsNeeded =
       await this.provider.selectCoinSetWithCombinedBalanceGreaterThanOrEqual(
         address,
-        BigInt(amount + DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER)
+        BigInt(amount + DEFAULT_GAS_BUDGET_FOR_SUI_TRANSFER),
```

Comment:
the default gas budget should be kept for sui coin type

Reply:
Fixed this